src/sycl/DataFormats/approx_atan2.h

```diff
@@ -0,0 +1,290 @@
+#ifndef DataFormatsMathAPPROX_ATAN2_H
+#define DataFormatsMathAPPROX_ATAN2_H
+
+/*
+ * approximate atan2 evaluations
+ *
+ * Polynomials were obtained using Sollya scripts (in comments below)
+ *
+ *
+*/
+
+/*
+f= atan((1-x)/(1+x))-atan(1);
+I=[-1+10^(-4);1.0];
+filename="atan.txt";
+print("") > filename;
+for deg from 3 to 11 do begin
+  p = fpminimax(f, deg,[|1,23...|],I, floating, absolute); 
+  display=decimal;
+  acc=floor(-log2(sup(supnorm(p, f, I, absolute, 2^(-20)))));
+  print( "   // degree = ", deg, 
+         "  => absolute accuracy is ",  acc, "bits" ) >> filename;
+  print("template<> constexpr float approx_atan2f_P<", deg, ">(float x){") >> filename;
+  display=hexadecimal;
+  print(" return ", horner(p) , ";") >> filename;
+  print("}") >> filename;
+end;
+*/
+
+#include <cstdint>
+#include <cmath>
+#include <limits>
+#include <algorithm>
+
+using sycl::abs;
```

Comment:
Does this `using` declaration rely on a file including this header to include also `CL/sycl.hpp`? Or do the SYCL compilers provide some magic?

Reply:
Yes to the first question, otherwise it wouldn't compile (but I think I've just been lucky and that's better to include `CL/sycl.hpp` here). 
The answer to the second question could be also a yes for what I've seen :)